src/graph/transform/cpu/knn.cc

```diff
@@ -15,6 +20,221 @@ namespace dgl {
 namespace transform {
 namespace impl {
 
+// This value is directly from pynndescent
+static constexpr int NN_DESCENT_BLOCK_SIZE = 16384;
```

Comment:
Can you justify this number? Like, what will happen if we increase or decrease it?

Reply:
We use batch update here. Thus we need some structures to record updates. If we increase this value, we may have a higher memory cost since we need to record more updates. If we decrease this value, we may have a higher time cost due to the small batch size.